src/main/java/uk/gov/justice/hmpps/prison/web/config/CacheConfig.java

```diff
@@ -41,62 +44,41 @@ public class CacheConfig implements CachingConfigurer {
     private int activityTimeoutSeconds;
 
-    @Bean(destroyMethod = "shutdown")
-    public net.sf.ehcache.CacheManager ehCacheManager() {
-        final var config = new net.sf.ehcache.config.Configuration();
-        config.sizeOfPolicy(new SizeOfPolicyConfiguration().maxDepth(20_000));
-
-        config.addCache(config("referenceDomain", 500, referenceDataTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("referenceCodesByDomain", 1000, referenceDataTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("referenceCodeByDomainAndCode", 1000, referenceDataTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-
-        config.addCache(config("caseNoteTypesByCaseLoadType", 100, caseNoteTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("getCaseNoteTypesWithSubTypesByCaseLoadTypeAndActiveFlag", 100, caseNoteTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("usedCaseNoteTypesWithSubTypes", 100, caseNoteTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-
-        config.addCache(config("findByStaffId", 1000, userTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("findRolesByUsername", 1000, userTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("loadUserByUsername", 5000, userTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-
-        config.addCache(config("findByStaffIdAndStaffUserType", 1000, userTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("findAgenciesByUsername", 1000, agencyTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config("findLocationsByAgencyAndType", 1000, locationTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-        config.addCache(config(GET_AGENCY_LOCATIONS_BOOKED, 500, activityTimeoutSeconds, MemoryStoreEvictionPolicy.LRU));
-
-        return net.sf.ehcache.CacheManager.newInstance(config);
-    }
-
-    public static CacheConfiguration config(final String name, final int maxElements, final int timeoutSeconds, final MemoryStoreEvictionPolicy policy) {
-        return new CacheConfiguration().name(name)
-                .memoryStoreEvictionPolicy(policy)
-                .eternal(false)
-                .overflowToOffHeap(false)
-                .maxEntriesLocalHeap(maxElements)
-                .timeToLiveSeconds(timeoutSeconds)
-                .timeToIdleSeconds(timeoutSeconds);
-    }
-
-    @Bean
-    @Override
-    public CacheManager cacheManager() {
-        return new EhCacheCacheManager(ehCacheManager());
-    }
-
     @Bean
-    @Override
-    public KeyGenerator keyGenerator() {
-        return new SimpleKeyGenerator();
+    public JCacheManagerCustomizer cacheConfiguration() {
+        return cm -> {
+            cm.createCache("referenceDomain", config(String.class, ReferenceDomain.class, 500, referenceDataTimeoutSeconds));
+            cm.enableStatistics("referenceDomain", true);
```

Comment:
hopefully this will generate the app insights custom metrics stuff for us instead of /info